tests/algorithms/pytorch/test_newton_raphson.py

```diff
@@ -49,33 +49,32 @@ def forward(self, x):
 @pytest.fixture(scope="module")
 def torch_algo(perceptron, numpy_torch_dataset):
     class MyAlgo(TorchNewtonRaphsonAlgo):
-        def __init__(self, model=None, criterion=None, batch_size=1, l2_coeff=0):
+        def __init__(self, model=None, dataset=None, criterion=None, batch_size=1, l2_coeff=0):
```

Comment:
why do you need to add a `dataset=None` argument? It seems like there is always `dataset=numpy_torch_dataset`

Reply:
I mean you never know if one would like to change dataset from time to time but I'll revert it as it is not the case anymore